app/src/main/java/com/example/drawingApp/MainActivity.kt

```diff
@@ -1,23 +1,111 @@
 package com.example.drawingApp
 
 import android.os.Bundle
+import android.os.StrictMode
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
 import androidx.activity.viewModels
+import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.widget.addTextChangedListener
 
 class MainActivity : AppCompatActivity() {
+    private val MAX_CHAR_LEN = 16
```

Comment:
Put constants like this in a companion object at the bottom

Reply:
created and moved